torch/_utils.py

```diff
@@ -446,7 +446,9 @@ def _get_device_attr(get_member):
 
 def _get_current_device_index():
     # current device index
-    return _get_device_attr(lambda m: m.current_device())
+    if torch.cuda.device_count() > 0:
+        return torch.cuda.current_device()
+    return -1
```

Comment:
This implementation isn't exactly the same as the old one for eager mode.

Reply:
Ok yes I agree. I don't think I considered all the cases this API might fail in eager mode earlier. I have restored the earlier eager mode change and instead used `torch.jit.is_scripting()` to check the mode and call the `JIT` helper API defined in `torch/jit/cuda.py`.